src/new/wallet.js

```diff
@@ -0,0 +1,439 @@
+   * server {String} Server for the wallet to connect to, e.g. http://localhost:8080/v1a/
+   * seed {String} 24 words separated by space
+   * passphrase {String} Wallet passphrase
+   * tokenUid {String} UID of the token to handle on this wallet
+   * password {String} Password to encrypt the seed
+   * pin {String} PIN to execute wallet actions
+   */
+  constructor({
+    network,
+    server = DEFAULT_SERVER,
+
+    seed,
+    passphrase = '',
+
+    tokenUid = HATHOR_TOKEN_CONFIG.uid,
+
+    // XXX Update it so we don't have fixed pin/password
+    password = '123',
+    pinCode = '123',
+  } = {}) {
+    super();
+
+    if (!network) {
+      throw Error('You must explicitly provide the network.');
+    }
+
+    if (!seed) {
+      throw Error('You must explicitly provide the seed.');
+    }
+
+    networkInstance.setNetwork(network);
+    this.network = network;
+
+    this.state = HathorWallet.CLOSED;
+    this.serverInfo = null;
+
+    this.onConnectionChange = this.onConnectionChange.bind(this);
+    this.handleWebsocketMsg = this.handleWebsocketMsg.bind(this);
+    this.onAddressesLoaded = this.onAddressesLoaded.bind(this);
+
+    this.seed = seed;
+    this.server = server;
+
+    // tokenUid is optional so we can get the token of the wallet
+    this.token = null;
+    this.tokenUid = tokenUid;
+
+    this.passphrase = passphrase;
+    this.pinCode = pinCode;
+    this.password = password;
+  }
+
+  /**
+   * Called when loading the history of transactions.
+   * It is called every HTTP Request to get the history of a set of addresses.
+   * Usually, this is called multiple times.
+   * The `historyTransactions` contains all transactions, including the ones from a previous call to this method.
+   *
+   * @param {Object} result {historyTransactions, allTokens, newSharedAddress, newSharedIndex, addressesFound}
+   **/
+  onAddressesLoaded(result) {
+    this.emit('more-addresses-loaded', result);
+  }
+
+  /**
+   * Called when the connection to the websocket changes.
+   * It is also called if the network is down.
+   **/
+  onConnectionChange(value) {
+    if (value) {
+      this.setState(HathorWallet.SYNCING);
+      wallet.loadAddressHistory(0, GAP_LIMIT).then(() => {
+        this.setState(HathorWallet.READY);
+      }).catch((error) => {
+        throw error;
+      })
+    } else {
+      // CONNECTING or CLOSED?
+      this.serverInfo = null;
+      this.setState(HathorWallet.CONNECTING);
+    }
+  }
+
+  getCurrentAddress({ markAsUsed = false } = {}) {
+    if (markAsUsed) {
+      return wallet.getAddressToUse();
+    }
+    return wallet.getCurrentAddress();
+  }
+
+  /**
+   * Called when a new message arrives from websocket.
+   **/
+  handleWebsocketMsg(wsData) {
+    if (wsData.type === 'wallet:address_history') {
+      this.onNewTx(wsData);
+    }
+  }
+
+  reloadData() {
+    // TODO Reload data?
+    console.log('reloadData');
+  }
+
+  getBalance(tokenUid) {
+    const uid = tokenUid || this.token.uid;
```

Comment:
Should use `this.tokenUid`.

Reply:
It's the same value.

Reply:
Ah, I thought we would not use `this.token` as per my other comment.